import-scripts/clear_cbioportal_persistence_cache.sh

```diff
@@ -43,34 +60,28 @@ function print_portal_id_values() {
     done
 }
 
-function authenticate_msk_service_account_if_necessary() {
-    if [ -z "KUBECONFIG_ARG" ] ; then
-        /data/portal-cron/scripts/authenticate_service_account.sh
-    fi
-}
-
 portal_id=$1
-KUBECONFIG_ARG=""
-if [ "$portal_id" == "public" ] || [ "$portal_id" == "genie-private" ] || [ "$portal_id" == "genie-archive" ] || [ "$portal_id" == "genie-public" ] ; then
-    if ! [ -z $PUBLIC_CLUSTER_KUBECONFIG ] ; then
-        KUBECONFIG_ARG="--kubeconfig $PUBLIC_CLUSTER_KUBECONFIG"
-    fi
-fi
-
 if [ -z "$portal_id" ] ; then
     echo "usage : $app_name <portal id>"
     print_portal_id_values
     exit 1
 fi
 
+cluster_id=${portal_to_cluster_map[$portal_id]}
 deployment_id=${portal_to_deployment_map[$portal_id]}
-if [ -z "$deployment_id" ] ; then
+if [ -z "$cluster_id" ] || [ -z "$deployment_id" ]; then
     echo "invalid portal_id : $portal_id"
     print_portal_id_values
     exit 1
 fi
 
-authenticate_msk_service_account_if_necessary
+unset KUBECONFIG_ARG
+if [ "$cluster_id" == "$CLUSTER_ID_DIGITS" ] ; then
+    /data/portal-cron/scripts/authenticate_service_account.sh
+else
+    KUBECONFIG_ARG="--kubeconfig $PUBLIC_CLUSTER_KUBECONFIG"
```

Comment:
keep the check to see if PUBLIC_CLUSTER_KUBECONFIG is set

Reply:
ah .. got it